File: demo-alianza-back/src/main/java/com/alianza/demo/entity/Person.java
package com.alianza.demo.entity;

import org.hibernate.annotations.GenericGenerator;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;


@Getter
@Setter
@Entity
public class Person extends BaseEntity{

    @Id
    @GeneratedValue(strategy= GenerationType.AUTO,generator="native")
    @GenericGenerator(name = "native",strategy = "native")
    private int personId;

    @NotBlank(message="sharedKey must not be blank")
    @Size(min=3, message="Name must be at least 3 characters long")
    private String sharedKey;
    
    @NotBlank(message="bussinesId must not be blank")
    @Size(min=3, message="Name must be at least 3 characters long")
    private String bussinesId;

    @NotBlank(message="Mobile number must not be blank")
    @Pattern(regexp="(^$|[0-9]{10})",message = "Mobile number must be 10 digits")
    private String mobileNumber;

    @NotBlank(message="Email must not be blank")
    @Email(message = "Please provide a valid email address" )
    private String email;


}

File: demo-alianza-back/src/js/src/app/common/person.ts
export class Person {

    personId?: number;
    sharedKey: string;
    bussinesId: string;
    mobileNumber: number;
    email: string;
    createdAt: string;
    updatedAt: string;

    

    constructor(sharedKey: string, bussinesId: string, mobileNumber: number, email: string, createdAt: string, updatedAt: string) {
        this.sharedKey = sharedKey;
        this.bussinesId = bussinesId;
        this.mobileNumber = mobileNumber;
        this.email = email;
        this.createdAt= createdAt;
        this.updatedAt= updatedAt;

    }
}